physics/lorenz_transformation_four_vector.py

```diff
@@ -0,0 +1,201 @@
+"""
+Lorenz transformation describes the transition from a reference frame P
+to another reference frame P', each of which is moving in a direction with
+respect to the other. The Lorenz transformation implemented in this code
+is the relativistic version using a four vector described by Minkowsky Space:
+x0 = ct, x1 = x, x2 = y, and x3 = z
+
+NOTE: Please note that x0 is c (speed of light) times t (time).
+
+So, the Lorenz transformation using a four vector is defined as:
+
+|ct'|   | γ  -γβ  0  0|  |ct|
+|x' | = |-γβ  γ   0  0| *|x |
+|y' |   | 0   0   1  0|  |y |
+|z' |   | 0   0   0  1|  |z |
+
+Where:
+           1
+γ = ---------------
+      -----------
+     /     v^2  |
+    /(1 -  ---
+  -/       c^2
+
+      v
+β = -----
+      c
+
+Reference: https://en.wikipedia.org/wiki/Lorentz_transformation
+"""
+from __future__ import annotations
+
+from math import sqrt
+
+import numpy as np  # type: ignore
+from sympy import symbols  # type: ignore
+
+# Coefficient
+# Speed of light (m/s)
+c = 299792458
+
+# Symbols
+ct, x, y, z = symbols("ct x y z")
+ct_p, x_p, y_p, z_p = symbols("ct' x' y' z'")
+
+
+# Vehicle's speed divided by speed of light (no units)
+def beta(velocity: float) -> float:
+    """
+    >>> beta(c)
+    1.0
+
+    >>> beta(199792458)
+    0.666435904801848
+
+    >> beta(1e5)
+    4.5031152851750526e-07
+
+    >>> beta(0.2)
+    Traceback (most recent call last):
+      ...
+    ValueError: Speed must be greater than 1!
+    """
+    if velocity > c:
+        raise ValueError("Speed must not exceed Light Speed 299,792,458 [m/s]!")
+
+    # Usually the speed u should be much higher than 1 (c order of magnitude)
+    elif velocity < 1:
+        raise ValueError("Speed must be greater than 1!")
+    return velocity / c
+
+
+def gamma(velocity: float) -> float:
+    """
+    >> gamma(4)
+    1.0000000000000002
+
+    >> gamma(1e5)
+    1.0000000556325075
+
+    >> gamma(3e7)
+    1.005044845777813
+
+    >> gamma(2.8e8)
+    2.7985595722318277
+
+    >> gamma(299792451)
+    4627.49902669495
+
+    >>> gamma(0.3)
+    Traceback (most recent call last):
+      ...
+    ValueError: Speed must be greater than 1!
+
+    >>> gamma(2*c)
+    Traceback (most recent call last):
+      ...
+    ValueError: Speed must not exceed Light Speed 299,792,458 [m/s]!
+    """
+    return 1 / (sqrt(1 - beta(velocity) ** 2))
+
+
+def transformation_matrix(velocity: float) -> np.array:
+    """
+    >> transformation_matrix(29979245)
```

Comment:
```suggestion
    >>> transformation_matrix(29979245)
```
3, not 2

Reply:
Thank you so much! 
Can't believe I missed that

Reply:
Some tests in the `beta` and `gamma` functions would require the same change, although the bot seems pleased.